merkle/smt/tile.go

```diff
@@ -37,6 +41,35 @@ type Tile struct {
 	Leaves []Node
 }
 
+// Merge returns a new tile which is a combination of this tile with the given
+// updates. The resulting tile contains all the nodes from the updates tile,
+// and all the nodes from the original tile not present in the updates.
+func (t Tile) Merge(updates Tile) (Tile, error) {
```

Comment:
Not sure that `Tile` is the best type for the parameter here. Logically, it's not a `Tile`, but a "delta". It just happens that this delta can be represented with the same type with the same invariants. WDYT?

Reply:
I actually think that some sort of `Updates` type could be used with other algorithms in this package, and would make them safer. E.g. [this](https://github.com/google/trillian/blob/d00b41283b9902ede843325991dd8e050cd3eeb8/merkle/smt/writer.go#L89) method (and `Split` method above it) could manipulate with `Updates` instead of `[]Node`.

Reply:
An `Updates` or `Delta` object which wraps/represents `[]Node` at the same level SGTM. The ID isn't necessary and can be inferred from the node IDs anyway.

Reply:
Will send a follow-up PR shortly.